src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  color: Color
+  shape: Shape
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
 
-export class Circle {
+    if (this.a <= 0
+      || this.b <= 0
+      || this.c <= 0
+      || this.a + this.b <= this.c
+      || this.b + this.c <= this.a
+      || this.c + this.a <= this.b) {
+      throw new Error('Wrong length of side');
+    }
+  }
 
+  getArea():number {
+    const halpP = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(halpP * (halpP - this.a)
+      * (halpP - this.b) * (halpP - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    this.shape = 'circle';
+
+    if (this.radius <= 0) {
+      throw new Error('Wrong radius');
+    }
+  }
 
+  getArea():number {
+    const area = Math.PI * (this.radius ** 2);
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    this.shape = 'rectangle';
+
+    if (this.width <= 0 || this.height <= 0) {
+      throw new Error('Wrong length of side');
```

Comment:
You should write why is it wrong. 
For example `Value can't be less than 1`
The same in all other errors